src/exoscale/compute/api/http.clj

```diff
@@ -164,10 +164,23 @@
           resp))
       resp)))
 
+(defn validate-job!!
+  [config opcode params]
+  (fn [{:keys [errortext errorcode] :as jobresult}]
+    (when (and (:throw-on-job-failure? config)
+               (some? errortext))
+      (throw (ex-info "Job failed"
+                      {:status errorcode
```

Comment:
why not return the `job-result` as data?

Reply:
Idea is to make async-job exceptions as close to other exceptions (generated in `http/with-decoded-error-body`)

Reply:
I have to double check how I create the exception.
Today we mostly throw _HTTP exceptions_.
Maybe there is a better way to do

Reply:
I create tests around request errors and async job failures to demonstrate that we handle both of them the same way.
I fixed an issue with `telex` integration. In case of errors, `telex` is throwing an exoscale.ex/ex-info with response associated to _response_ key in ex-data (Aleph used to put response as the ex-data).